_posts/2018-08-19-openfaas-on-fargate.md

```diff
@@ -0,0 +1,169 @@
+
+
+To run OpenFaaS in it's recommended configuration we need to schedule 5 containers: `gateway`, `nats`, `prometheus`, 
+`prometheus alert manager` and a platform provider in this case `faas-fargate`
+
+Cost per day $2.28 (note this does not include compute costs for actually running any functions!, more on that later)
+
+
+#### EKS
+[EKS Pricing](https://aws.amazon.com/fargate/pricing/)
+[EC2 Pricing](https://aws.amazon.com/ec2/pricing/)
+
+| Region              |[us-east-1](https://docs.aws.amazon.com/general/latest/gr/rande.html) |
+|:------------------|:------------|
+| Cost of cluster per hour:  |   $0.20 |
+| EC2 on-demand cost per hour:</br>Kubernetes worker node |  $0.1<br/><small>assuming instance size of m4.large</small>           |
+| Cost per day: <br/> <small>with one worker node | **$7.20** <br/><small>=24 * (0.2 + 0.1)</small>|
+
+
+> For small deployments, running on Fargate will be significantly cheaper.
+
+## Architecture
+OpenFaaS functions are built as docker images and deployed as docker containers. This approach will be very familiar to
+developer using Microservices. The platform is composed of a `gateway`, `provider` and `monitoring` components. The 
+gateway delegates the deployment and invoking of functions to the `provider`. The provider in our case if `faas-fargate`,
+which has been specifically designed to run and invoke functions using Fargate containers.
+
+The diagram below shows how each of these components interact:
+
+![openfaas-fargate](/images/openfaas-on-fargate/openfaas-fargate-component-architecture.png)
+
+Each of the components are created and deployed as Fargate container.
+
+Architecture features:
+
+* ** \[security\] ** we restrict traffic between containers using [aws security groups](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html)
+* ** \[security\] ** isolate ingress traffic, i.e. from the internet, in a `public subnet`. Container workloads are 
+scheduled in a `private subnet`
+* ** \[discovery\] ** all workloads use [Amazon Route 53 auto naming](https://docs.aws.amazon.com/Route53/latest/APIReference/overview-service-discovery.html)
+to manage service registration / de-registration
+* ** \[logging\] ** we configure each container workload to log to [aws cloudwatch].
+
+## Running OpenFaaS on Fargate
+
+The easiest way to run [OpenFaaS on Fargate](https://github.com/ewilde/faas-fargate) is to use the [terraform module](https://github.com/ewilde/terraform-aws-openfaas-fargate)
+I developed to deploy all the necessary components:
+
+ * [OpenFaaS gateway](https://github.com/openfaas/faas)
+ * [FaaS-Fargate provider](https://github.com/ewilde/faas-fargate) 
+ * [Prometheus](https://github.com/prometheus/prometheus)
+ * [Prometheus alert manager](https://github.com/prometheus/alertmanager)
+ * [NATS messaging server](https://github.com/nats-io/gnatsd)
+
+Simply run the following commands. **Costs will be incurred**:
```

Comment:
Can you be more specific `**Costs will be incurred**`? Maybe add it as a disclaimer instead?

I'd add a quick section on the end on how to tear-down the setup too.

Reply:
added tear down section and disclaimer